src/Persistence_matrix/include/gudhi/Fields/Z2_field.h

```diff
@@ -0,0 +1,355 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hannah Schreiber
+ *
+ *    Copyright (C) 2022-24 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+/**
+ * @file Z2_field.h
+ * @author Hannah Schreiber
+ * @brief Contains the @ref Z2_field_element class.
+ */
+
+#ifndef MATRIX_FIELD_Z2_H_
+#define MATRIX_FIELD_Z2_H_
+
+#include <utility>
+
+namespace Gudhi {
+namespace persistence_fields {
+
+/**
+ * @class Z2_field_element Z2_field.h gudhi/Fields/Z2_field.h
+ * @ingroup persistence_fields
+ *
+ * @brief Class representing an element of the \f$ \mathbb{F}_2 \f$ field.
+ */
+class Z2_field_element
+{
+ public:
+  using element_type = bool;  /**< Type for the elements in the field. */
+  template <class T>
+  using isInteger = std::enable_if_t<std::is_integral_v<T> >;
```

Comment:
It was public first to make it appear in the documentation (as it is used in public templates). But at the end, it was more confusing than helping, so I decided to not document it. It remained public because it seemed harmless. Would it be cleaner if I make it private? (also in the other classes)